### PR TITLE
Full Project Design and Interactions

### PR DESCRIPTION
**Pull request by Obimbo07**
This pull request introduces the following changes:

- Projects Navigation components and full design.
- The Rockets Components with booking interactions.
- The Mission Components with Join & leave Mission interactions.
- Dragons Components implemented with booking interactions.
- My Profile section with reserved rockets and dragons displayed. The joined missions are also displayed here.
- Components unit tests.

### CODE REVIEW

**Review by franclobo — COMMENTED** [review]
Hi team, 

# Invalid Code Review Request

descarga

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't have a professional README file, please use the Microverse template.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/components/MyProfile.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  }
+
+  let renderReservedMissions = '';
+  if (reservedMissionTitles.length) {
+    renderReservedMissions = reservedMissionTitles.map((title) => (
+      <li className="filter-list" key={title}>
+        <span>{title}</span>
+      </li>
+    ));
+  }
```
Comment (AbayomiOlaoye) [inline src/components/MyProfile.js]: - [ ] What a great of `ES6 map()` here and it really worked just fine except that the requirement asked that you opt for `filter()` instead.

image
- To match up with the above, could you use `filter()` for your filtering?

File: src/redux/dragonSlice.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        state.data = action.payload;
+      })
+      .addCase(fetchdragons.rejected, (state) => {
+        state.isLoading = false;
+      });
+  },
+});
+export const getReservedDragons = (state) => state.dragon.reserved;
+export const { reserveDragon } = dragonSlice.actions;
+export default dragonSlice.reducer;
```
Comment (AbayomiOlaoye) [inline src/redux/dragonSlice.js]: - [ ] Your reducers are working just fine but for more organization and readability, the requirement suggested that you create a `directory` for each of these redux state slice files, e.g. `dragons`, `missions`, and `rockets`.

image

**Review by AbayomiOlaoye — CHANGES_REQUESTED** [review]
## STATUS: CHANGE REQUIRED ♻️ 

Hi @Team :wave: :wave: , 

Good job with this milestone so far! What a brilliant attempt!!
There are just a few more issues to work on before moving on to the next project but you are almost there!

so_close

## Highlights
- Descriptive Pull Request message :heavy_check_mark: 
- Good use of GitHub flow :heavy_check_mark: 
- Meaningful commit messages :heavy_check_mark: 
- No linter error :heavy_check_mark: 
- Store was correctly set :heavy_check_mark:
- All tests passing :heavy_check_mark: 
- Use of ES6/ES7 syntax :heavy_check_mark:
- Descriptive ReadMe documentation :heavy_check_mark: 

## Required Changes ♻️
- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TadesseDev — APPROVED** [review]
### Hi Team

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
AmazingCreepyFrogmouth-max-1mb




## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅


## Optional suggestions

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**